### PR TITLE
feat: implement overflow when producing

### PR DESCRIPTION
**Pull request by thalesmg**
Addresses https://emqx.atlassian.net/browse/EMQX-8300

Also bumps `replayq` -> 0.3.5 (fix for ignoring corrupt segments)

### CODE REVIEW

**Comment by thalesmg** [comment]
Will re-tag `0.7.1` after merge.

File: src/pulsar_producer.erl [hunk 1]
```diff
@@ -702,7 +706,21 @@ enqueue_send_requests(Requests, State = #{replayq := Q}) ->
...
+
+-spec handle_overflow(state(), integer()) -> state().
+handle_overflow(State, Overflow) when Overflow =< 0 ->
+    %% no overflow
+    State;
+handle_overflow(State0 = #{replayq := Q, callback := Callback}, Overflow) ->
+    {NewQ, QAckRef, Items0} =
+        replayq:pop(Q, #{bytes_limit => Overflow, count_limit => 999999999}),
+    ok = replayq:ack(NewQ, QAckRef),
+    log_warn("replayq dropped ~b overflowed messages", [length(Items0)], State0),
```
Comment (zmstone) [inline src/pulsar_producer.erl:720]:
```suggestion
    log_warn("replayq dropped ~b overflown messages", [length(Items0)], State0),
```

**Review by zmstone — APPROVED** [review]

File: src/pulsar_producer.erl (hunk 1 above)
Comment (thalesmg) [reply src/pulsar_producer.erl:720]: "Overflown" would be the past participle of "overfly", it seems.

**Comment by thalesmg** [comment]
`0.7.1` re-tagged: 1bf19a0381664ac15d560339d8dd5dd8c7e985fd

**Pull request merged**